File: kernel/drivers/native/kbd.lua
local BaseChrDev = require("drivers.basechrdev")

---@class Kbd: BaseChrDev
local Kbd = BaseChrDev:new()

function Kbd:push_event(event)
    table.insert(self.queue, event)
end

function Kbd:read_file(inode, offset, length)
	return table.remove(self.queue, 1)
end

return Kbd

File: kernel/drivers/native/kbd.ts
import { Inode } from "../../vfs/inode/init";
import { BaseChrDev } from "../common/basechrdev";

export class Kbd extends BaseChrDev {
	event_queue: string[] = [];

	push_event(event: string) {
		this.event_queue.push(event);
	}

	read_file(inode: Inode, offset: integer, length: integer): string {
		return this.event_queue.shift() || "";
	}

	write_file(inode: Inode, offset: integer, data: string): integer {
		return 0;
	}

	ioctl(inode: Inode, request: integer, arg: any): unknown {
		return;
	}
}
